src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs

```diff
@@ -267,15 +294,25 @@ public class AccessController : Controller
                 });
             }
 
+            var request = HttpContext.GetOpenIdConnectRequest();
```

Comment:
> You mean, a parameter?

It's not really a parameter, at least not _stricto sensu_: `OpenIdConnectRequest` is populated by ASOS as part of the request extraction process, using either the query string or the request body (depending on the type of the request and whether GET or POST was used). It's then stored in a special feature named `OpenIdConnectServerFeature` that holds both the OIDC request and the response objects. `GetOpenIdConnectRequest()` is a wrapper around that stuff.

> Whatever you think is less ugly but would reuse the BadRequest code and message.

`BadRequest` is only used here ('cause it's an "API action"). The other actions return a view when something is wrong.